pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -91,6 +91,9 @@ type DiscoveryServer struct {
 	// APIs and service registry info
 	ConfigGenerator core.ConfigGenerator
 
+	// Generators allow customizing the generated config, based on the client metadata.
+	Generators map[string]model.XdsResourceGenerator
```

Comment:
what is the key?

Reply:
Good question, will update the docs. Next iteration will be a bit more flexibile in the key.